sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/ControlFlowGraph/UcfgInstructionBuilder.cs

```diff
@@ -85,123 +85,154 @@ public Instruction Create(SyntaxNode syntaxNode)
             }
         }
 
-        public IEnumerable<Instruction> CreateAttributeInstructions(AttributeSyntax attributeSyntax, IMethodSymbol attributeCtor, string parameterName) =>
+        public IEnumerable<Instruction> CreateAttributeInstructions(AttributeSyntax attributeSyntax, IMethodSymbol attributeCtor,
+            string parameterName) =>
             new[]
             {
-                CreateMethodCall(attributeSyntax, UcfgMethod.Create(attributeCtor)),
+                AddMethodCall(attributeSyntax, attributeCtor),
                 CreateParameterAnnotation(parameterName, attributeSyntax),
             };
 
-        private Instruction CreateDefaultInstruction(SyntaxNode node) =>
-            CreateConstant(node);
+        private IEnumerable<Instruction> CreateDefaultInstruction(SyntaxNode node) =>
+            new[] { CreateConstant(node) };
```

Comment:
CreateConstant could return Enumerable.Empty and you will not be required to create a new array here and in many other places